### ISSUE TITLE
Aborted using `cfg-if`: memory allocation of <tons of> bytes failed

### ISSUE THREAD

**Issue by caoweinian** [rust-lang/rustfmt#5428]
origin: https://internals.rust-lang.org/t/cargo-fmt-failed-with-infinite-memory-request/16950

Using `cargo fmt` to format the following code causes the process to be aborted with one line error message: "memory allocation of &lt;tons of&gt; bytes failed":

```rust
// cfg-if: version 1.0.0
cfg_if::cfg_if! {
    if #[cfg(windows)] {
        compile_error!{"..."};
    }
}
```

rustc/cargo: 1.62.0
rustfmt: 1.4.38-stable

**Comment by ytmimi** [comment]
Thanks for reaching out.

I believe this may be a duplicate of https://github.com/rust-lang/rustfmt/issues/4442. I just checked and this looks to be resolved by #5341, which is still under review.

**Comment by ytmimi** [comment]
going to close this as a duplicate of https://github.com/rust-lang/rustfmt/issues/4442. I'd follow that issue or PR https://github.com/rust-lang/rustfmt/pull/5341 to see when the fix is merged.
